lib/coqProject_file.ml

```diff
@@ -151,7 +151,8 @@ let check_filename f =
   let a = ref None in
   let check_char c =
     match c with
-    | '\n' | '\r' | '\t' | '\\' | '\'' | '"' | ' ' | '#' | '$' | '%' -> a := Some c
+    | '\\' | '$' when Sys.os_type = "Unix" -> a := Some c
```

Comment:
What happens on macos?

Reply:
I took out `$` for now, that likely won't come up too much on Windows.  Very hard to test this because the dune build doesn't have a coq_makefile-prelude and  building a modified version for Windows is a project (I rarely do this anymore).  If it's not a fix, it is at least a step in the right direction.